doc/compiler/aot/Diagnostics.md

```diff
@@ -0,0 +1,396 @@
+         eb4b0d2  44   TR_InlinedVirtualMethod (55)   4     Abs       eb219040
+ Removed Guard inlined method: Inlined site index = 1, Constant pool = dd47a870, cpIndex = 78, romClassOffsetInSharedCache=0000000000042F70
+...
+         eb4b3f6  10   TR_HelperAddress (1)           2     Rel 20    0024
+Helper method address of j2iTransition(20)
+         eb4b400  18   TR_HCR (41)                    2     Abs       0012 HCR: patch location offset = 00000000000AAAE0
+         eb4b412  12   TR_RamMethod (21)              2     Abs       0012 0242
+</relocatableDataCG>
+```
+
+#### Header
+```
+Code start = 14de6fc0, Method start pc = 14de7000, Method start pc offset = 0x40
+java/util/concurrent/ConcurrentHashMap.tabAt([Ljava/util/concurrent/ConcurrentHashMap$Node;I)Ljava/util/concurrent/ConcurrentHashMap$Node;
+```
+This section contains information about the method being compiled.
+It is worth noting that `Code start` here does not mean the address
+of the method when it is loaded into the Code Cache; this is the
+address in the temporarily allocated memory in the Code Cache used
+for the purpose of compilation. At the end of compilation, the method
+is stored into the SCC and loaded at a later point.
+
+#### Relocation Record Generation Info
+```
+Type                                File                             Line  Offset(M) Offset(PC) Node
+TR_InlinedVirtualMethodWithNopGuard (34) /root/hostdir/openj9-openjdk-jdk11/openj9/runtime/compiler/codegen/J9CodeGenerator.cpp  2774      007e       003e 0000000000000000
+TargetAddress1:00007F470F23B980,  TargetAddress2:0000000000000000
+
+```
+This section contains information about the abstract notion of the
+relocation record, namely the `TR::ExternalRelocationRecord`. Objects
+of this type are iterated over, and used to generate the actual
+relocation records that are stored into the SCC, described by the
+`TR_RelocationRecordHelperAddressBinaryTemplate` family of structs.
+
+Here, `TR_InlinedVirtualMethodWithNopGuard` is the relocation type,
+specifically the `TR_ExternalRelocationTargetKind` enum; `34` is the
+value of the enum. Also printed is the file and line number where this
+External Relocation was generated. Next follows two offsets, the first
+is the offset from the start of the allocated memory for this method,
+and the second is the offset from the startPC. This is followed by the
+`TR::Node` that is associated with this relocation record; this is used
+to connect the relocation record with the IL. In the case of validation
+or inlined site records, this may be NULL; however for the
+former the node does not matter and for the latter, the inlined site
+index can be determined from the next section below. Finally, the two
+values passed into the constructor of the `TR::ExternalRelocationRecord`
+is printed out; these values are used to generate the values in the
+actual relocation record that gets stored into the SCC.
```

Comment:
I wonder if this paragraph could be organized in bulletin points to make it easier to spot each item 

- Type 
   - `TR_InlinedVirtualMethodWithNopGuard` is the relocation type ... 
- File & Line
   - The file and line number where this External Relocation was generated ... 
- Offset(M)
   - The offset from the start of the allocated memory for this method
- Offset(PC)
   - The offset from the startPC
- Node
  - The TR::Node that is associated with this relocation record ... 
- TargetAddress1 & TargetAddress2 
  - ...

Reply:
Update in force push.